api/lib/validate.js

```diff
@@ -166,8 +178,95 @@ function validateOrderbook(orderbook) {
+    if (settings.transfer_rate !== '') {
+      throw error('Parameter must be a number: transfer_rate');
+    }
+  }
+  if (!/(undefined|number)/.test(typeof settings.wallet_size)) {
+    if (settings.wallet_size !== '') {
+      throw error('Parameter must be a number: wallet_size');
+    }
+  }
+  if (!/(undefined|boolean)/.test(typeof settings.no_freeze)) {
+    throw error('Parameter must be a boolean: no_freeze');
+  }
+  if (!/(undefined|boolean)/.test(typeof settings.global_freeze)) {
+    throw error('Parameter must be a boolean: global_freeze');
+  }
+  if (!/(undefined|boolean)/.test(typeof settings.password_spent)) {
+    throw error('Parameter must be a boolean: password_spent');
+  }
+  if (!/(undefined|boolean)/.test(typeof settings.disable_master)) {
+    throw error('Parameter must be a boolean: disable_master');
+  }
+  if (!/(undefined|boolean)/.test(typeof settings.require_destination_tag)) {
+    throw error('Parameter must be a boolean: require_destination_tag');
+  }
+  if (!/(undefined|boolean)/.test(typeof settings.require_authorization)) {
+    throw error('Parameter must be a boolean: require_authorization');
+  }
+  if (!/(undefined|boolean)/.test(typeof settings.disallow_xrp)) {
+    throw error('Parameter must be a boolean: disallow_xrp');
+  }
+
+  var setCollision = (typeof settings.no_freeze === 'boolean')
+    && (typeof settings.global_freeze === 'boolean')
+    && settings.no_freeze === settings.global_freeze;
+
+  if (setCollision) {
+    throw error('Unable to set/clear no_freeze and global_freeze');
+  }
+}
+
 function validateTrustline(trustline) {
-  validateSchema(trustline, 'Trustline');
+  if (typeof trustline !== 'object') {
+    throw error('Invalid parameter: trustline');
+  }
+  if (_.isUndefined(trustline.limit)) {
+    throw error('Parameter missing: trustline.limit');
+  }
+  if (isNaN(trustline.limit)) {
+    throw error('Parameter is not a number: trustline.limit');
+  }
+  if (!trustline.currency) {
+    throw error('Parameter missing: trustline.currency');
+  }
+  if (!validator.isValid(trustline.currency, 'Currency')) {
+    throw error('Parameter is not a valid currency: trustline.currency');
+  }
+  if (!trustline.counterparty) {
+    throw error('Parameter missing: trustline.counterparty');
+  }
+  if (!isValidAddress(trustline.counterparty)) {
+    throw error('Parameter is not a Ripple address: trustline.counterparty');
+  }
+  if (!/^(undefined|number)$/.test(typeof trustline.quality_in)) {
+    throw error('Parameter must be a number: trustline.quality_in');
+  }
+  if (!/^(undefined|number)$/.test(typeof trustline.quality_out)) {
+    throw error('Parameter must be a number: trustline.quality_out');
+  }
+  if (!/^(undefined|boolean)$/.test(typeof trustline.account_allows_rippling)) {
+    throw error('Parameter must be a boolean: trustline.allow_rippling');
+  }
+  // TODO: validateSchema(trustline, 'Trustline');
 }
 
```

Comment:
Just a thought - given the amount of repetition and the need to validate many items, we've used this library with great success and convenience: https://www.npmjs.com/package/checkit

Just to put it on your radar in case it is preferred.